### ISSUE TITLE
Alacritty power up disabled graphic card on a laptop

### ISSUE THREAD

**Issue by Flowneee** [alacritty/alacritty#7526]
After upgrading to `0.13` version on startup alacritty, presumably, scans all graphic adapters, powering them up. In my case I have an Nvidia discrete card, which most of the time in D3 state, powered up only when I need it (for games for example). In my case this cause lag at startup, because card require time to power up, and also starts fans.

Such behavior was not present (at least for me) on `0.12` version.

### System

OS: Arch Linux
Version: alacritty 0.13.0 (78fa4d6f)
Linux/BSD: X11, i3 + picom

### Logs

`0.13` (card powered down before start)

```
$ cat /sys/bus/pci/devices/0000\:01\:00.0/power_state; alacritty -v
D3cold
Created log file at "/tmp/Alacritty-11603.log"
[0.000002092s] [INFO ] [alacritty] Welcome to Alacritty
[0.000027516s] [INFO ] [alacritty] Version 0.13.0 (78fa4d6f)
[0.000030603s] [INFO ] [alacritty] Running on X11
[0.000094325s] [INFO ] [alacritty] Configuration files loaded from:
                                     "/home/flowneee/.config/alacritty/alacritty.toml"
[0.002558060s] [INFO ] [alacritty] Using GLX 1.4
[1.416373271s] [INFO ] [alacritty] Window scale factor: 1.5
[1.432743597s] [INFO ] [alacritty] Running on Mesa Intel(R) Graphics (ADL GT2)
[1.432759515s] [INFO ] [alacritty] OpenGL version 4.6 (Core Profile) Mesa 23.3.2-arch1.2, shader_version 4.60
[1.432762193s] [INFO ] [alacritty] Using OpenGL 3.3 renderer
[1.443527555s] [INFO ] [alacritty] Cell size: 14 x 29
[1.443548246s] [INFO ] [alacritty] Padding: 0 x 0
[1.443550749s] [INFO ] [alacritty] Width: 800, Height: 600
[1.445230903s] [INFO ] [alacritty] PTY dimensions: 20 x 57
[1.447467961s] [INFO ] [alacritty] Initialisation complete
[1.454764477s] [INFO ] [alacritty] Padding: 0 x 0
[1.454782106s] [INFO ] [alacritty] Width: 1274, Height: 1533
```

`0.13.0` (card is powered up already)

```
$ cat /sys/bus/pci/devices/0000\:01\:00.0/power_state; alacritty -v
D0
Created log file at "/tmp/Alacritty-11642.log"
[0.000002375s] [INFO ] [alacritty] Welcome to Alacritty
[0.000032239s] [INFO ] [alacritty] Version 0.13.0 (78fa4d6f)
[0.000036053s] [INFO ] [alacritty] Running on X11
[0.000108813s] [INFO ] [alacritty] Configuration files loaded from:
                                     "/home/flowneee/.config/alacritty/alacritty.toml"
[0.002338595s] [INFO ] [alacritty] Using GLX 1.4
[0.039028020s] [INFO ] [alacritty] Window scale factor: 1.5
[0.055863131s] [INFO ] [alacritty] Running on Mesa Intel(R) Graphics (ADL GT2)
[0.055877665s] [INFO ] [alacritty] OpenGL version 4.6 (Core Profile) Mesa 23.3.2-arch1.2, shader_version 4.60
[0.055880245s] [INFO ] [alacritty] Using OpenGL 3.3 renderer
[0.067105372s] [INFO ] [alacritty] Cell size: 14 x 29
[0.067127715s] [INFO ] [alacritty] Padding: 0 x 0
[0.067130011s] [INFO ] [alacritty] Width: 800, Height: 600
[0.072236633s] [INFO ] [alacritty] PTY dimensions: 20 x 57
[0.074588273s] [INFO ] [alacritty] Initialisation complete
[0.082448639s] [INFO ] [alacritty] Padding: 0 x 0
[0.082474801s] [INFO ] [alacritty] Width: 1274, Height: 1533
```

`0.12.3` (card powered down before start)

```
$ cat /sys/bus/pci/devices/0000\:01\:00.0/power_state; alacritty -v
D3cold
Created log file at "/tmp/Alacritty-12181.log"
[0.000004541s] [INFO ] [alacritty] Welcome to Alacritty
[0.000029588s] [INFO ] [alacritty] Version 0.12.3 (5efb0690)
[0.000032500s] [INFO ] [alacritty] Running on X11
[0.002214977s] [INFO ] [alacritty] Configuration files loaded from:
                                     "/home/flowneee/.config/alacritty/alacritty.yml"
[0.004002333s] [INFO ] [alacritty] Using GLX 1.4
[0.026431595s] [INFO ] [alacritty] Window scale factor: 1.5
[0.041895710s] [INFO ] [alacritty] Running on Mesa Intel(R) Graphics (ADL GT2)
[0.041910735s] [INFO ] [alacritty] Using OpenGL 3.3 renderer
[0.063653888s] [INFO ] [alacritty] Cell size: 14 x 29
[0.063672643s] [INFO ] [alacritty] Padding: 0 x 0
[0.063674675s] [INFO ] [alacritty] Width: 800, Height: 600
[0.065354125s] [INFO ] [alacritty] PTY dimensions: 20 x 57
[0.067573056s] [INFO ] [alacritty] Initialisation complete
[0.075673719s] [INFO ] [alacritty] Padding: 0 x 0
[0.075697322s] [INFO ] [alacritty] Width: 1274, Height: 1533
```

**Comment by kchibisov** [comment]
We have no control over GPUs and no control what device we end up picking, because, OpenGL.

If you want to force specific GPU, use `libglvnd` for that.

**Comment by Flowneee** [comment]
Found "bad" commit: 80d4daccc307622f8e604af4f36ac89ecccf5db7

**Comment by Flowneee** [comment]
> We have no control over GPUs and no control what device we end up picking, because, OpenGL.

I'm going to disagree. Problem is not selecting wrong device, it still running on correct card. Problem is something accessing card in a way that powering it up, which it shouldn't normally do. Otherwise I would have such problem with any GUI application. I assume that it has to do with new dependencies, 

```diff
diff --git a/alacritty/Cargo.toml b/alacritty/Cargo.toml
index 44a8bc2f..0752b2d2 100644
--- a/alacritty/Cargo.toml
+++ b/alacritty/Cargo.toml
@@ -29 +29 @@ clap = { version = "4.2.7", features = ["derive", "env"] }
-copypasta = { version = "0.8.1", default-features = false }
+copypasta = { version = "0.10.0", default-features = false }
@@ -31 +31 @@ crossfont = { version = "0.5.0", features = ["force_system_fontconfig"] }
-glutin = { version = "0.30.4", default-features = false, features = ["egl", "wgl"] }
+glutin = { version = "0.31.0", default-features = false, features = ["egl", "wgl"] }
@@ -37,0 +38 @@ parking_lot = "0.12.0"
+raw-window-handle = "0.5"
@@ -43 +44 @@ unicode-width = "0.1"
-winit = { version = "0.29.1-beta", default-features = false, features = ["serde"] }
+winit = { version = "0.29.2", default-features = false, features = ["rwh_05", "serde"] }
@@ -61,3 +61,0 @@ objc = "0.2.2"
-[target.'cfg(not(any(target_os="windows", target_os="macos")))'.dependencies]
-x11-dl = { version = "2", optional = true }
-
@@ -83 +80,0 @@ x11 = [
-    "x11-dl",
```

**Comment by Flowneee** [comment]
Seems like it is related to upgrading `winit`. I checked all other dependencies, they cause no problems.

**Comment by kchibisov** [comment]
And winit doesn't touch GPU stuff at all, so how should we suppose to fix it?

Also, have you tried latest master?

**Comment by Flowneee** [comment]
> And winit doesn't touch GPU stuff at all

Maybe it could be related to glutin-winit, IDK, I just reported problem, unfortunately I have no deep knowledge about crates in question.

> Also, have you tried latest master?

Thanks for reminding about that. Checked, problem is not present, seems like I should stick to current version an wait for next major release, current is bugged for me.

I'm going to close this issue then, @kchibisov thank you for quick response.
